### ISSUE TITLE
TAGGING_POLICY=OpticalOnly tags everything

### ISSUE THREAD

**Issue by Davood-Norouzi** [broadinstitute/picard#1952]
Hi,

I am using picard 2.18.29 with bioconda.

I have a fastq/bam file with more than 18% duplicates (no PCR involved). It consists of 2% regular duplicates which I am interested in, but also has 16% optical duplicates which I am trying to remove.

I tried this:

>> picard MarkDuplicates  I=sorted_in.bam  O=dups_NoOpt.bam  M=matrix_NoOpt.txt  REMOVE_DUPLICATES=false TAGGING_POLICY=OpticalOnly

Then
>> samtools view -b -F 0x400  dups_NoOpt.bam > filtered.bam

Then re-did the duplicate tagging for the remaining duplicates:
>> picard MarkDuplicates  I=filtered.bam  O=dups_MarkD.bam  M=matrix_MarkD.txt  REMOVE_DUPLICATES=false

The final matrix_MarkD.txt shows both READ_PAIR_DUPLICATES and READ_PAIR_OPTICAL_DUPLICATES are zero.

Why? What am I doing wrong? I though I am only removing the optical duplicates.

Another question: when using picard with bioconda, how to pass the options on how many threads and how much memory picard should utilize?

Thanks
-Davood

**Comment by kockan** [comment]
I don't believe the SAM flag differentiates between optical and non-optical duplicates. If the read is determined to be a duplicate, no matter the type, that flag will be set. The tagging policy adds additional SAM tags (the tag should be `DT`) to reads so you could try performing your filtering based on these tags and see if it works.

**Comment by Davood-Norouzi** [comment]
@kockan could you please elaborate on how to remove only optical duplicates using the tag that you mentioned? Thank you!

**Comment by kockan** [comment]
Hmm, let's say you set `TAGGING_POLICY=OpticalOnly` as before. Then, the value of the `DT` tag should be `SQ` for optical duplicates and due to your selection, the `DT` tag shouldn't appear in any other reads, including library duplicates.
With this you might be able to do something like:

`samtools view -e '![DT]' input.bam > DT.removed.bam`

Best to check the samtools manual page (https://www.htslib.org/doc/samtools-view.html) to make sure you get the command-line right though, I haven't tested this myself.

**Comment by Davood-Norouzi** [comment]
Thanks a lot @kockan!
